fdbserver/Ratekeeper.actor.cpp

```diff
@@ -309,6 +309,42 @@ class RatekeeperImpl {
 		}
 	}
 
+	ACTOR static Future<Void> monitorHotShards(Ratekeeper* self, Reference<AsyncVar<ServerDBInfo> const> dbInfo) {
+		loop {
+			wait(delay(SERVER_KNOBS->SERVER_LIST_DELAY));
+			if (!self->ssHighWriteQueue.size()) {
+				continue;
+			}
+
+			loop {
+				state SetThrottledShardRequest setReq;
+				state UID ssi;
+				state int i = 0;
+
+				ssi = self->ssHighWriteQueue[i];
+
+				TraceEvent(SevDebug, "SendGetBusyShardsRequest");
+				try {
+					GetBusyShardsRequest getReq;
+					GetBusyShardsReply reply = wait(self->storageServerInterfaces[ssi].getBusyShards.getReply(getReq));
```

Comment:
We can do better here by parallelizing requests to all IDs in `ssHighWriteQueue`.